compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1400,13 +1400,23 @@ static LogicalResult setRootConfig(
   return success();
 }
 
+namespace {
+bool isPoolingOp(linalg::LinalgOp op) {
```

Comment:
A lot of this would be much simpler if we added a `PoolingInterface` upstream (just like `ConvolutionInterface`).  would also be OK with just making the `pooling` ops implement the `ConvolutionInterface`.

Reply:
They do implement the ConvolutionInterface, but the tiling code seems to be handling only 2-dimensional cases. I replicated exactly the same heuristics for pooling.